test cases/warning/10 cython limited api version/meson.build

```diff
@@ -0,0 +1,12 @@
+project('cython limited api version', ['c', 'cython'])
+
+cythonc = meson.get_compiler('cython')
+if cythonc.version() >= '3.0'
+  error('MESON_SKIP_TEST: Cython compiler supports this functionality, no warning necessary')
```

Comment:
I'm not sure it's valuable to have a test case testing that we emit a warning. This is running an additional project cycle just to verify the UX, but it's not obvious to me that the cython version check is something that is at risk of being gotten incorrect.

Reply:
I'll remove this.